tests/test_md.py

```diff
@@ -907,3 +915,99 @@ def build_context(barostat_interval):
     gc.collect()
     for ref in reffed_objs:
         assert ref() is None
+
+
+def test_local_md_nonbonded_all_pairs_subset():
+    """Test that if the nonbonded all pairs is set up on a subset of the system, that local MD can correctly
+    simulate the local region without double counting interactions"""
+    seed = 2023
+
+    temperature = constants.DEFAULT_TEMP
+    dt = 1.5e-3
+    friction = 1.0
+
+    mol, _ = get_biphenyl()
+    ff = Forcefield.load_from_file("smirnoff_1_1_0_sc.py")
+
+    # Lambda must either be 1.0 (uninteracting) or minimize energy, chose 1.0 as cheaper to not-minimize
+    # else will overflow and coordinates will be different between tests
+    unbound_potentials, sys_params, masses, coords, box = get_solvent_phase_system(mol, ff, 1.0, minimize_energy=False)
+
+    identity_idxs = np.arange(0, len(coords), dtype=np.int32)
+    ligand_idxs = np.arange(len(coords) - mol.GetNumAtoms(), len(coords), dtype=np.int32)
+    non_ligand_idxs = np.delete(identity_idxs, ligand_idxs)
+
+    bound_pots = [bp.bind(params) for bp, params in zip(unbound_potentials, sys_params)]
+
+    summed_bound = next(fn for fn in bound_pots if isinstance(fn.potential, SummedPotential))
+    nb_pot_idx = next(i for i, pot in enumerate(summed_bound.potential.potentials) if isinstance(pot, Nonbonded))
+    nb_pot = summed_bound.potential.potentials[nb_pot_idx]
+    nb_bound = nb_pot.bind(summed_bound.potential.params_init[nb_pot_idx])
```

Comment:
nit: maybe slightly cleaner to use `zip` here:

```suggestion
    nb_bound = next(
        pot.bind(params)
        for pot, params in zip(summed_bound.potential.potentials, summed_bound.potential.params_init)
        if isinstance(pot, Nonbonded)
    )
```

Reply:
Need nb_pot to construct the other potential and this pushes more logic than I like in a single line.

Reply:
Ah, fair enough (I updated the suggestion to expose both `nb_pot` and `nb_params`, but agreed that it's a matter of preference)